napari_aicssegmentation/controller/workflow_steps_controller.py

```diff
@@ -31,9 +33,24 @@ def close_workflow(self):
         self.model.reset()
         self.router.workflow_selection()
 
-    def run_all(self):
-        result = self.model.active_workflow.execute_all()
-        self.viewer.add_image(result, name=f"Result for workflow {self.model.active_workflow.name}")
+    def run_all(self, parameter_inputs: List[Dict[str, List]]):
+        """ 
+        parameter_inputs is a list of dictionaries, one dictionary (or None if no parameters) 
+        for each step. Each dictionary has the same shape as a WorkflowStep.parameter_defaults
+        dictionary, but with the parameter values obtained from the UI instead of default values.
+        """
+
+        workflow = self.model.active_workflow
+
+        # This is equivalent to running workflow.execute_all(), but with non-default params
+        workflow.reset()
+        step = 0
```

Comment:
This should start at 1- we copy the initial layer as layer 0

Reply:
but dont worry about it, my pr should take care of running each step and naming them correctly

Reply:
She's iterating on workflow steps, not on Layers